src/app/Settings/CredentialsStorage.tsx

```diff
@@ -0,0 +1,121 @@
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { Select, SelectOption, SelectVariant, Text } from '@patternfly/react-core';
+
+import { UserSetting } from './Settings';
+import { getFromLocalStorage, saveToLocalStorage } from '@app/utils/LocalStorage';
+
+export interface Location {
+  key: string;
+  description: string;
+}
+
+export class Locations {
+  static readonly BROWSER_SESSION: Location = {
+    key: 'Session (Browser Memory)',
+    description:
+      'Keep credentials in browser memory for the current session only. When you close this browser tab the credentials will be forgotten.',
+  };
+  static readonly BACKEND: Location = {
+    key: 'Backend',
+    description:
+      'Keep credentials in encrypted Cryostat backend storage. These credentials will be available to other users and will be used for Automated Rules.',
+  };
+}
+
+const locations = [Locations.BROWSER_SESSION, Locations.BACKEND];
+
+function getLocation(key: string): Location {
+  for (let l of locations) {
+    if (l.key === key) {
+      return l;
+    }
+  }
+  return Locations.BROWSER_SESSION;
+}
+
+const Component = () => {
+  const [isExpanded, setExpanded] = React.useState(false);
+  const [selection, setSelection] = React.useState(Locations.BROWSER_SESSION);
+
+  const handleSelect = React.useCallback(
+    (_, selection) => {
+      let location = getLocation(selection);
+      setSelection(location);
+      setExpanded(false);
+      saveToLocalStorage('JMX_CREDENTIAL_LOCATION', selection);
+    },
+    [getLocation, setSelection, setExpanded, saveToLocalStorage]
+  );
+
+  React.useEffect(() => {
+    handleSelect(undefined, getFromLocalStorage('JMX_CREDENTIAL_LOCATION', Locations.BROWSER_SESSION));
+  }, [handleSelect, getLocation, getFromLocalStorage]);
```

Comment:
Does getLocation need to be a dep here? From my understanding handleSelect already has getLocation as a dep so the handleSelect function would change anyways, but I'm not completely sure.

Reply:
Oh yea, thanks. This used to actually call `getLocation` before I cleaned things up and opened the PR. You're right that it isn't needed here anymore.